components/vocabularies/orphanet/api/src/main/java/org/phenotips/vocabulary/internal/solr/Orphanet.java

```diff
@@ -0,0 +1,215 @@
+        // orphanet label == phenotips vocab name
+        doc.addField("id", getOrphanetId(ontClass.getLocalName()));
+        doc.addField("name", ontClass.getLabel(null));
+
+        // save all parents
+        List<OntClass> parents = ontClass.listSuperClasses().toList();
+        for (OntClass parent : parents) {
+            if (parent.isRestriction() && parent.asRestriction().isSomeValuesFromRestriction()) {
+                // a restriction, we only care about "SomeValuesFrom" restrictions
+                // get get the value of the restriction, make sure it's under the roots we care about, not some other
+                // property.
+                // TODO in another iteration, later:
+                //   epidemiology (1/2000000, family, etc)
+                //   prevalence
+                Restriction restriction = parent.asRestriction();
+                OntClass value = restriction.asSomeValuesFromRestriction().getSomeValuesFrom().as(OntClass.class);
+                for (OntClass root : roots) {
+                    if (value.hasSuperClass(root)) {
+                        doc.addField("term_category", getOrphanetId(value.getLocalName()));
+                    }
+                }
+            } else if (isRegularClass(parent)) {
+                // just a regular clas:s.
+                doc.addField("term_category", getOrphanetId(parent.getLocalName()));
+                // we should be able to use the {{@link OntClass.getSuperClass()}} method, but since it returns an
+                // **arbitrary** selection, we'll just pick the first superclass that's actually a class
+                // that we stumble upon.
+                if (doc.getField("is_a") == null) {
+                    doc.addField("is_a", getOrphanetId(parent.getLocalName()));
+                }
+            }
+        }
+
+        extractProperties(doc, ontClass);
+        return doc;
+    }
+
+    private boolean isRegularClass(OntClass ontClass) {
+        return !(ontClass.isRestriction()
+                || ontClass.isEnumeratedClass()
+                || ontClass.isUnionClass()
+                || ontClass.isIntersectionClass()
+                || ontClass.isComplementClass());
+    }
+
+    private void extractProperties(SolrInputDocument doc, OntClass ontClass) {
+        List<Statement> statements = ontClass.listProperties().toList();
+        for (Statement statement : statements) {
+            Resource subject = statement.getSubject();
+            RDFNode object = statement.getObject();
+            Property predicate = statement.getPredicate();
+
+            if (!subject.as(OntClass.class).equals(ontClass)) {
+                throw new RuntimeException("Subject is not the disease.");
+            }
+
+            String relation = predicate.getLocalName();
+            switch (relation) {
+                case "hasDbXref":
+                    extractDBXref(doc, object);
+                    break;
+                case "alternative_term":
```

Comment:
hmm perhaps. This is a mapping from Orphanet's OWL properties to Solr fields, which is not 1-to-1 for most fields, except for `alternative_term`. I'll add a constant to the class for Solr field names.